docs/virtual_machines/instancetypes.md

```diff
@@ -0,0 +1,298 @@
+* [HostDevices](http://kubevirt.io/api-reference/main/definitions.html#_v1_hostdevice) : Optional list of HostDevices to passthrough
+* IOThreadsPolicy : Optional IOThreadsPolicy to be used
+* [LaunchSecurity](http://kubevirt.io/api-reference/main/definitions.html#_v1_launchsecurity): Optional LaunchSecurity to be used
+
+Anything provided within an instancetype cannot be overridden within the [`VirtualMachine`](http://kubevirt.io/api-reference/main/definitions.html#_v1_virtualmachine). For example as `CPU` and `Memory` are both required attributes of an instancetype if a user makes any requests for `CPU` or `Memory` resources within the underlying [`VirtualMachine`](http://kubevirt.io/api-reference/main/definitions.html#_v1_virtualmachine) the instancetype will conflict and the request will be rejected during creation.
+
+## VirtualMachinePreference
+
+```yaml
+---
+apiVersion: instancetype.kubevirt.io/v1alpha1
+kind: VirtualMachinePreference
+metadata:
+  name: example-preference
+spec:
+  devices:
+    preferredDiskBus: virtio
+    preferredInterfaceModel: virtio
+```
+
+KubeVirt also provides two further preference based [`CRDs`](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/), again a cluster wide [`VirtualMachineClusterPreference`](https://kubevirt.io/api-reference/main/definitions.html#_v1_virtualmachineclusterpreference) and namespaced [`VirtualMachinePreference`](https://kubevirt.io/api-reference/main/definitions.html#_v1_virtualmachinepreference). These [`CRDs`](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) encapsulate the preferred value of any remaining attributes of a [`VirtualMachine`](http://kubevirt.io/api-reference/main/definitions.html#_v1_virtualmachine) required to run a given workload, again this is through a shared [`VirtualMachinePreferenceSpec`](http://kubevirt.io/api-reference/main/definitions.html#_v1_virtualmachinepreferencespec).
+
+Unlike instancetypes preferences only represent the preferred values and as such can be overridden by values in the [`VirtualMachine`](http://kubevirt.io/api-reference/main/definitions.html#_v1_virtualmachine) provided by the user.
+
+For example as shown below, if a user has provided a [`VirtualMachine`](http://kubevirt.io/api-reference/main/definitions.html#_v1_virtualmachine) with a disk bus already defined within a [DiskTarget](http://kubevirt.io/api-reference/main/definitions.html#_v1_disktarget) *and* has also selected a set of preferences with [DevicePreference](http://kubevirt.io/api-reference/main/definitions.html#_v1_devicepreferences) and `preferredDiskBus` defined the users original choice within the [`VirtualMachine`](http://kubevirt.io/api-reference/main/definitions.html#_v1_virtualmachine) and [DiskTarget](http://kubevirt.io/api-reference/main/definitions.html#_v1_disktarget) are used:
```

Comment:
The example is already using the VirtualMachinePreferenceMatcher, which will be introduced in the next section. But I have no idea how to avoid it.

Reply:
I wanted to introduce the types before introducing how to reference them from the `VirtualMachine`,